src/main/java/org/primefaces/extensions/component/letteravatar/LetterAvatarRenderer.java

```diff
@@ -71,18 +95,56 @@ public void encodeMarkup(FacesContext context, LetterAvatar letterAvatar) throws
             writer.writeAttribute("style", style, "style");
         }
 
-        writer.writeAttribute("width", width, null);
-        writer.writeAttribute("height", height, null);
-        writer.writeAttribute("avatar", avatar, null);
+        writer.writeAttribute("title", value, null);
+
+        writer.startElement("span", letterAvatar);
+        writer.writeAttribute("class", "ui-letteravatar-initials", null);
+        writer.writeAttribute("style", styleSpan(size), null);
+        writer.write(initials);
+        writer.endElement("span");
+
+        writer.endElement("div");
+    }
 
-        writer.endElement("img");
+    private int hash(String str) {
+        int result = 0;
+        for (int i = 0; i < str.length(); i++) {
+            result = ((result << 5) - result) + str.charAt(i);
+            result |= 0;
+        }
+        return result;
+    }
+
+    private int hue(String str) {
+        return Math.abs(hash(str) % 360);
+    }
+
+    private String styleDiv(String size, String color, String backgroundColor, boolean rounded) {
+        Map<String, String> map = new LinkedHashMap<>();
+        map.put("color", color);
+        map.put("background-color", backgroundColor);
+        if (rounded) {
+            map.put("border-radius", "50%");
+        }
+        map.put("height", size);
+        map.put("text-align", "center");
+        map.put("width", size);
+        return map.entrySet()
```

Comment:
This could be DRYed by adding a method

Reply:
@jepsar its all checked into master now if you want to make any tweaks!

Reply:
Alright. I'll have a look.